includes/Core/Util/User_Input_Settings.php

```diff
@@ -126,19 +126,27 @@ function( $setting ) {
 	 * @return array|WP_Error User input settings.
 	 */
 	private function sync_with_proxy( $settings = null ) {
-		$user_input_settings_url  = $this->authentication->get_google_proxy()->url( Google_Proxy::USER_INPUT_SETTINGS_URI );
+		$user_input_settings_url = $this->authentication->get_google_proxy()->url( Google_Proxy::USER_INPUT_SETTINGS_URI );
+		$creds                   = $this->authentication->credentials()->get();
+
 		$user_input_settings_args = array(
 			'headers' => array(
 				'Authorization' => 'Bearer ' . $this->authentication->get_oauth_client()->get_access_token(),
+				'Content-Type'  => 'application/json',
+			),
+			'body'    => array(
+				'site_id'     => $creds['oauth2_client_id'],
+				'site_secret' => $creds['oauth2_client_secret'],
 			),
 		);
 
 		if ( ! empty( $settings ) ) {
-			$user_input_settings_args['headers']['Content-Type'] = 'application/json';
-			$user_input_settings_args['body']                    = wp_json_encode( $settings );
+			$user_input_settings_args['body']['settings']       = $settings;
+			$user_input_settings_args['body']['client_user_id'] = get_current_user_id();
 		}
 
```

Comment:
before sending the request, the body should be JSON encoded. ie add:
`$user_input_settings_args['body'] = wp_json_encode( $user_input_settings_args['body'] );`

Reply:
Ah that's correct, good catch. Did you try leaving out the call to `wp_json_encode` entirely? I'm wondering whether the WP request mechanism would automatically JSON-encode the body with `Content-Type` set to `application/json`. Probably not, but worth trying.

Reply:
Without json encoding the body, I got an "unmarshaling" error. I don't think we have encoding for body in core's `WP_Http` class [request](https://github.com/WordPress/wordpress-develop/blob/8eda4984e1d409a97b260c52b3485defccae495d/src/wp-includes/class-http.php#L149)

Reply:
This caused me to get a `{"error":"missing authentication token"}` error repeatedly in testing. I now have it working.